src/Microsoft.DotNet.CodeFormatting.Tests/Analyzers/ExplicitVariableTypeAnalyzerTests.cs

```diff
@@ -0,0 +1,237 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.CodeFormatting.Analyzers;
+
+using Xunit;
+
+namespace Microsoft.DotNet.CodeFormatting.Tests
+{
+    public sealed class ExplicitVariableTypeAnalyzerTests : AnalyzerFixerTestBase
+    {
+        public ExplicitVariableTypeAnalyzerTests()
+        {
+            DisableAllDiagnostics();
+            EnableDiagnostic(ExplicitVariableTypeAnalyzer.DiagnosticId);
+        }
+        [Fact]
+        public void TestSimpleTypeExplicitDeclaration()
+        {
+            var text = @"
+class C1
+{
+    void M()
+    {
+        int x = 0;
+        bool[] y = { true };  
+        int[][] z = new[] { new[] { 1, 2, 3 }, new[] { 4, 5, 6 } };
+    }
+}";
+            var expected = @"
+class C1
+{
+    void M()
+    {
+        int x = 0;
+        bool[] y = { true };  
+        int[][] z = new[] { new[] { 1, 2, 3 }, new[] { 4, 5, 6 } };
+    }
+}";
+            Verify(text, expected, runFormatter: false);
+        }
+
+        [Fact]
+        public void TestSimpleTypeVarDeclaration()
+        {
+            var text = @"
+class C1
+{
+    void M()
+    {
+        var x = 0;
+        var y = new[] { true, false };  
+        var z = new[] { new[] { 1, 2, 3 }, new[] { 4, 5, 6 } };
+        var anon = new { Name = ""Terry"", Age = 34 };
+    }
+}";
+            var expected = @"
+class C1
+{
+    void M()
+    {
+        Int32 x = 0;
+        Boolean[] y = new[] { true, false };  
+        Int32[][] z = new[] { new[] { 1, 2, 3 }, new[] { 4, 5, 6 } };
+        var anon = new { Name = ""Terry"", Age = 34 };
```

Comment:
I think for clarity this last one is a separate test: TestAnonymousTypeDeclaration.